### PR TITLE
Add nightly-images workflow

### LINKED ISSUE #84: Move nightly build job to istio-ecosystem repo
The job is currently executed [in the fork](https://github.com/openshift-service-mesh/sail-operator) because that was easy to setup - credentials for pushing images and the GitHub bot account were already present. We should move this job to this repository though.

### PR DESCRIPTION
**Pull request by bmangoen**
#### What type of PR is this?

- [X] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [ ] Test
- [ ] Documentation Update

#### What this PR does / why we need it:

Moves the nightly image builds into a GitHub Action

#### Which issue(s) this PR fixes:

Fixes #84

### CODE REVIEW

**Comment by FilipB** [comment]
I guess we should prepare a PR which is removing the [job](https://github.com/openshift/release/blob/master/ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml#L381) running in our fork before merging this.

**Comment by bmangoen** [comment]
> I guess we should prepare a PR which is removing the [job](https://github.com/openshift/release/blob/master/ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml#L381) running in our fork before merging this.

@FilipB Yes the followiung [PR](https://github.com/openshift/release/pull/58104) is waiting for review as well

**Review by FilipB — APPROVED** [review]

**Pull request merged**
